### PR TITLE
Release work for two server releases

### LINKED ISSUE #1144: [RELEASE] MinIO Server RELEASE.2024-03-03T17-50-39Z
MinIO Server [RELEASE.2024-03-03T17-50-39Z](https://github.com/minio/minio/releases/tag/RELEASE.2024-03-03T17-50-39Z) has changes that impact the docs:

- [x] `userCredentials` setting added for NATS notifications [PR #19139](https://github.com/minio/minio/pull/19139)
- [x] With the latest KES, root credentials can be automatically generated if a KMS is enabled [PR #19025](https://github.com/minio/minio/pull/19025)
- [x] Adds new configs for expiring tiered objects [PR #18926](https://github.com/minio/minio/pull/18926)

### LINKED ISSUE #1155: [RELEASE] Server  RELEASE.2024-03-10T02-53-48Z
MinIO Server [RELEASE.2024-03-10T02-53-48Z](https://github.com/minio/minio/releases/tag/RELEASE.2024-03-10T02-53-48Z) has changes that impact the docs:

- [x] New webhooks config/envvar options [PR #19214](https://github.com/minio/minio/pull/19214)

### PR DESCRIPTION
**Pull request by djwfyi**
Release work for Server RELEASE.2024-03-03T17-50-39Z
    
- Adds new NATS config
- Adds new ilm configs
- Adds info on having a KMS auto generate root credentials
 
Closes #1144

Release work for Server RELEASE.2024-03-10T02-53-48Z
    
- Corrects a missing URL excludes line for some builds
- Adds new webhook options
    
Closes #1155

Staged:
- [Webhook config](http://192.241.195.202:9000/staging/release-work/linux/reference/minio-server/settings/metrics-and-logging.html#envvar.MINIO_AUDIT_WEBHOOK_BATCH_SIZE)
- [NATS config](http://192.241.195.202:9000/staging/release-work/linux/reference/minio-server/settings/notifications/nats.html#user-credentials-file)
- [Root credentials](http://192.241.195.202:9000/staging/release-work/linux/operations/install-deploy-manage/deploy-minio-single-node-single-drive.html#create-the-environment-variable-file)
- [ILM config](http://192.241.195.202:9000/staging/release-work/linux/reference/minio-server/settings/ilm.html)

### CODE REVIEW

**Comment by djwfyi** [comment]
Need to add the root credentials to other install pages.

File: source/includes/common/common-deploy.rst [hunk 1]
```diff
@@ -32,7 +32,25 @@ The following example provides a starting environment file:
 
    #MINIO_SERVER_URL="http://minio.example.net:9000"
 
-Include any other environment variables as required for your local deployment.
+Include any other environment variables as required for your deployment.
+
+.. versionadded:: Server RELEASE.2024-03-03T17-50-39Z
+
+   You can have MinIO automatically generate root credentials if all of the following are true:
```
Comment (feorlen) [inline source/includes/common/common-deploy.rst:39]: "Can" ? Seems like "will" if the prerequisites are met. Or is there a way to disable the automagical creds if you didn't want them? (Whether you should or not might be another story.)

File: source/includes/common/common-deploy.rst [hunk 2]
```diff
@@ -32,7 +32,25 @@ The following example provides a starting environment file:
...
+     - ``MINIO_ROOT_USER`` variable 
+     - ``MINIO_ROOT_PASSWORD`` variable 
+  
+   - **Have**:
+     - set up KES with a :kes-docs:`supported KMS target <#supported-kms-targets>`
+     - disabled root access with the :ref:`MinIO environment variable <minio-disable-root-access>`
+
+   When those conditions are met at startup, MinIO uses the KMS to generate unique root credentials for the deployment using a `hash-based message authentication code (HMAC) <https://en.wikipedia.org/wiki/HMAC>`__.
+
+   If MinIO generates such credentials, the key used to generate the credentials **must** remain the same *and* continue to exist.
```
Comment (feorlen) [inline source/includes/common/common-deploy.rst:52]: What happens if you need to rotate it?

Comment (aead) [reply source/includes/common/common-deploy.rst:52]: You can use a different key at the KMS. For example, you start with `MINIO_KMS_KES_KEY_NAME=my-key`. This key is used to generate the root credentials. After some time you want/have to change the credentials. You simply change `MINIO_KMS_KES_KEY_NAME=my-key-2`. Now, the new key is used to generate the root credentials.

File: source/includes/common/common-deploy.rst (hunk 1 above)
Comment (aead) [reply source/includes/common/common-deploy.rst:39]: If the conditions are meet, MinIO will always generate the credentials auto.
Of course, you can simply opt-out by setting the credentials yourself.

Comment (djwfyi) [reply source/includes/common/common-deploy.rst:39]: Rephrased

**Review by feorlen — APPROVED** [review]

**Pull request merged**